### ISSUE TITLE
Override test duration

### ISSUE THREAD

**Issue by ornatwork** [xunit/xunit#1469]
I'm doing a bit of cheating as explained in issue #1461, I start a process that runs a bunch of RSpec tests returning results when done, things work as expected.  Except that the runner thinks rightfully that my tests are each taking less than 1 millisecond to execute as all the results are available after RSpec finishes and are released as fast as possible.  Is there a way for me to override the ExecutionTime / Duration property from my test code ?  Or do I have to create my own test runner to intercept, as [suggested here.](https://stackoverflow.com/questions/29778512/xunit-net-capture-the-result-of-each-test-right-after-it-runs)

**Comment by bradwilson** [comment]
You would probably start by overriding `[Fact]` and `XunitTestCase`, and find the appropriate place (down the hierarchy of runners, starting with `XunitTestCaseRunner`) to override the execution and record the results w/ appropriate times. The [Partial Trust example](https://github.com/xunit/samples.xunit/tree/master/PartialTrustExample) overrides are done like this, and can show you how they're strung together.

**Comment by ornatwork** [comment]
That works, I created sample project and I'm able to override the test time for [Fact here](https://github.com/deloitte-solvas/SlackerRunnerSample/commit/b1c99f2c2823589dd7dc7f744040e79bb6d42ccc#diff-fa59cc1f768907e86cca1a851654f0be).

However I need this for `[Theory]` nor for `[Fact]`.  From reading the xunit code, I'm assuming I can in similar way implement `SlackerRunnerDataDiscoverer : DataDiscoverer` and `SlackerTheoryTestCaseRunner : XunitTheoryTestCaseRunner` ...

**Comment by bradwilson** [comment]
Yes, that's where you can start and see if there is sufficient hooks to do what you want to do.

**Comment by ornatwork** [comment]
I was barking up the wrong tree, with `DataDiscoverer` I found `TheoryDiscoverer `and implemented `SlackerTheoryDiscoverer, SlackerTheoryTestCase, SlackerTheoryTestCaseRunner` .  However it isn't working correctly, not sure why at this point.  Looks like things work during discovery but throws error when run.

I found it, apparently it's my override in 
[SlackerTheoryTestCaseRunner.RunTestAsync](https://github.com/deloitte-solvas/SlackerRunnerSample/blob/master/SlackerRunner/Xunit/XunitExtensions/SlackerTheoryTestCaseRunner.cs)

any suggestions ?

**Comment by bradwilson** [comment]
I don't see anything wrong on the surface. What's happening?

**Comment by ornatwork** [comment]
3 tests found during discovery
![discover](https://user-images.githubusercontent.com/94155/30996412-f61455bc-a47d-11e7-9caa-0aff520505fc.jpg)

Only 1 test run during the test run
![run](https://user-images.githubusercontent.com/94155/30996414-fab23968-a47d-11e7-8ff7-cc1e5731f0e1.jpg)

**Comment by bradwilson** [comment]
Is this working yet? Sorry, it fell through the cracks.

**Comment by ornatwork** [comment]
No, not yet, there is something wrong with the override code in [RunTestAsync](https://github.com/deloitte-solvas/SlackerRunnerSample/blob/master/SlackerRunner/Xunit/XunitExtensions/SlackerTheoryTestCaseRunner.cs#L51)  when I comment out the RunTestAsync funciton things work as expected.

**Comment by bradwilson** [comment]
Hello,

As part of the effort to clean up older issues, this is being closed due to age. If this is still applicable with v3, please open a new issue (with an attached v3 repro) so it can be re-evaluated and categorized.

Thanks!
